packages/fleather/lib/src/widgets/history.dart

```diff
@@ -213,6 +197,34 @@ class HistoryStack {
     }
     return null;
   }
+
+  /// Returns true if there is at least one redo operation.
+  bool get canRedo {
+    return _list.isNotEmpty && (_currentIndex < _list.length - 1);
+  }
+
+  static TextSelection selectionFromDelta(Delta changeDelta) {
```

Comment:
We should have this as a public static method.
See comment in `editor_toolbar.dart`